client/driver/docker.go

```diff
@@ -347,9 +347,8 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 		// TODO refactor the implementation in GetTaskEnv to match
 		// the 0.2 ports world view. Docker seems to be the only place where
 		// this is actually needed, but this is kinda hacky.
```

Comment:
Not sure what to do with this comment. What is the 0.2 ports world view? A port map is now passed to the taskEnv so that it can use this when building out the ports env.